src/common/sec_websocket_extensions.rs

```diff
@@ -0,0 +1,369 @@
+/// extension-token = registered-token
+/// registered-token = token
+/// extension-param = token [ "=" (token | quoted-string) ]
+/// ```
+///
+/// ## Example Values
+///
+/// * `permessage-deflate` (defined in [RFC7692][RFC7692_7])
+/// * `permessage-deflate; server_max_window_bits=10`
+/// * `permessage-deflate; server_max_window_bits=10, permessage-deflate`
+///
+/// ## Example
+///
+/// ```rust
+/// # extern crate headers;
+/// use headers::SecWebsocketExtensions;
+///
+/// let extensions = SecWebsocketExtensions::from_static("permessage-deflate");
+/// ```
+///
+/// ## Splitting and Combining
+///
+/// Note that `Sec-WebSocket-Extensions` may be split or combined across multiple headers.
+/// The following are equivalent:
+/// ```text
+/// Sec-WebSocket-Extensions: foo
+/// Sec-WebSocket-Extensions: bar; baz=2
+/// ```
+/// ```text
+/// Sec-WebSocket-Extensions: foo, bar; baz=2
+/// ```
+///
+/// `SecWebsocketExtensions` splits extensions when decoding and combines them into a single
+/// value when encoding.
+///
+/// [RFC6455_11.3.2]: https://tools.ietf.org/html/rfc6455#section-11.3.2
+/// [RFC7692_7]: https://tools.ietf.org/html/rfc7692#section-7
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SecWebsocketExtensions(pub Vec<WebsocketExtension>);
+
+impl Header for SecWebsocketExtensions {
+    fn name() -> &'static ::HeaderName {
+        &SEC_WEBSOCKET_EXTENSIONS
+    }
+
+    fn decode<'i, I: Iterator<Item = &'i HeaderValue>>(values: &mut I) -> Result<Self, Error> {
+        let extensions = values
+            .cloned()
+            .flat_map(|v| {
+                FlatCsv::<Comma>::from(v)
+                    .iter()
+                    .map(WebsocketExtension::try_from)
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        if extensions.is_empty() {
+            Err(Error::invalid())
+        } else {
+            Ok(SecWebsocketExtensions(extensions))
+        }
+    }
+
+    fn encode<E: Extend<HeaderValue>>(&self, values: &mut E) {
+        if !self.is_empty() {
+            values.extend(std::iter::once(self.to_value()));
+        }
+    }
+}
+
+impl SecWebsocketExtensions {
+    /// Construct a `SecWebSocketExtensions` from a static string.
+    ///
+    /// ## Panic
+    ///
+    /// Panics if the static string is not a valid extensions valie.
+    pub fn from_static(s: &'static str) -> Self {
+        let value = HeaderValue::from_static(s);
+        SecWebsocketExtensions::try_from(&value).expect("valid static string")
+    }
+
+    /// Convert this `SecWebsocketExtensions` to a single `HeaderValue`.
+    pub fn to_value(&self) -> HeaderValue {
+        let values = self.0.iter().map(HeaderValue::from).collect::<FlatCsv>();
+        HeaderValue::from(&values)
+    }
+
+    /// An iterator over the `WebsocketExtension`s in `SecWebsocketExtensions` header(s).
+    pub fn iter(&self) -> impl Iterator<Item = &WebsocketExtension> {
+        self.0.iter()
```

Comment:
Similar question here, what if instead of owning a `Vec`, the return iterator were lazy, and did the parsing as requested on each call to `next`, like other headers?